### DISCUSSION TITLE
Animating Transparency?

### DISCUSSION THREAD

**Discussion by MaggleMitch** [iimachines/Maya2glTF#165] [Q&A]
I was just wondering if it's possible to have keyed transparency exported from Maya so I can fade elements in and out etc.

**Comment by ziriax** [comment]
Unfortunately the glTF spec doesn't support transparency animation :-(

**Reply by MaggleMitch** [reply]
Damn - no worries. Does the same go for ‘visibility’ too? I haven’t been able to get that to work so have tried to mimic that by setting scale to 0 but it doesn’t always translate well. That’s how I planned to do sprite sheet animation

**Reply by MaggleMitch** [reply]
I've sorted this - found this issue (from here: https://github.com/iimachines/Maya2glTF/issues/152) and have gotten the scale working fine by making the keys stepped and throwing "-detectStepAnimations 2" in the extra flags box. Thanks!

**Comment by ziriax** [comment]
`-dsa 2` should work in such a case, it was designed for it.

If not, you might have found a bug

**Reply by MaggleMitch** [reply]
Is there a way to get image sequence texture keys coming through? Or would you basically need to do the above method on a number of different objects to mimic the image sequence rather than on one object?

**Reply by ziriax** [reply]
Unfortunately glTF 2.0 doesn't support texture animation.

https://stackoverflow.com/questions/55806088/gltf-2-0-video-texture-support

It seems your scene requires a more advanced format? 

glTF is really simple; you might want to look at e.g. USD, but I have no experience with that. 

It is indeed unfortunate that glTF doesn't support "animate everything" like Maya does. That would require a dataflow aka dependency graph, like Maya.
